source/Handlebars.Test/ViewEngine/ViewEngineTests.cs

```diff
@@ -147,7 +149,212 @@ public void CanIgnoreCommentsContainingHtml()
+                                              Compatibility =
+                                              {
+                                                  SupportLastInObjectIterations = true,
+                                              },
+                                          };
+            var handlebars = Handlebars.Create(handlebarsConfiguration);
+            var render = handlebars.CompileView("views\\someview.hbs");
+            var output = render(
+                new Dictionary<string, object>
+                {
+                    { "Foo", "Bar" },
+                    { "Baz", "Foo" },
+                    { "Bar", "Baz" },
+                }
+            );
+
+            Assert.Equal("Layout: First:Foo=Bar;Baz=Foo;Last:Bar=Baz; View", output);
+        }
+
+        [Fact]
+        public void CanIterateOverObjectInLayout()
+        {
+            var files = new FakeFileSystem
+                        {
+                            { "views\\layout.hbs", "Layout: {{#each this}}{{#if @First}}First:{{/if}}{{#if @Last}}Last:{{/if}}{{@Key}}={{@Value}};{{/each}}{{{body}}}" },
+                            { "views\\someview.hbs", "{{!< layout}} View" },
+                        };
+
+            var handlebarsConfiguration = new HandlebarsConfiguration
+                                          {
+                                              FileSystem = files,
+                                              Compatibility =
+                                              {
+                                                  SupportLastInObjectIterations = true,
+                                              },
+                                          };
+            var handlebars = Handlebars.Create(handlebarsConfiguration);
+            var render = handlebars.CompileView("views\\someview.hbs");
+            var output = render(
+                new
+                {
+                    Foo = "Bar",
+                    Baz = "Foo",
+                    Bar = "Baz",
+                }
+            );
+
+            Assert.Equal("Layout: First:Foo=Bar;Baz=Foo;Last:Bar=Baz; View", output);
+        }
+
+        [Fact]
+        public void CanIterateOverEnumerableInLayout()
+        {
+            var files = new FakeFileSystem
+                        {
+                            { "views\\layout.hbs", "Layout: {{#each this}}{{#if @First}}First:{{/if}}{{#if @Last}}Last:{{/if}}{{this}};{{/each}}{{{body}}}" },
+                            { "views\\someview.hbs", "{{!< layout}} View" },
+                        };
+
+            var handlebarsConfiguration = new HandlebarsConfiguration
+                                          {
+                                              FileSystem = files,
+                                              Compatibility =
+                                              {
+                                                  SupportLastInObjectIterations = true,
```

Comment:
nit: This flag is not used here

Reply:
I actually 'needed' to do this, because otherwise `{{@Last}}` is never true. Because of this: https://github.com/Handlebars-Net/Handlebars.Net/blob/master/source/Handlebars/ValueProviders/ObjectIteratorValues.cs#L66

Reply:
But in this case you're iterating over `IEnumerable`, don't you?

Reply:
You're absolutely correct. I blindly copied this from the other tests, assuming I would also need it there. Fixed.